htdocs/mrp/class/mo.class.php

```diff
@@ -706,6 +709,7 @@ public function updateProduction(User $user, $notrigger = true)
 							if ($moline->qty <= 0) {
 								$error++;
 								$this->error = "BadValueForquantityToConsume";
+								dol_print_error(0, $langs->trans("BadValueForquantityToConsume", $moline->qty));
```

Comment:
The mo class can be executed in any context. dol_print_error is allowed into web page context only
On error, class must return a negative value after setting ->error property.